framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -213,23 +261,22 @@ public void updateForMethodCall(
             boolean sideEffectsUnrefineAliases =
                     ((GenericAnnotatedTypeFactory) atypeFactory).sideEffectsUnrefineAliases;
 
-            // update local variables
             // TODO: Also remove if any element/argument to the annotation is not
             // isUnmodifiableByOtherCode.  Example: @KeyFor("valueThatCanBeMutated").
             if (sideEffectsUnrefineAliases) {
-                localVariableValues
-                        .entrySet()
-                        .removeIf(e -> !e.getKey().isUnmodifiableByOtherCode());
-            }
-
-            // update this value
-            if (sideEffectsUnrefineAliases) {
-                thisValue = null;
-            }
-
-            // update field values
-            if (sideEffectsUnrefineAliases) {
-                fieldValues.entrySet().removeIf(e -> !e.getKey().isUnmodifiableByOtherCode());
+                if (!sideEffectsOnlyExpressions.isEmpty()) {
+                    for (JavaExpression e : sideEffectsOnlyExpressions) {
+                        if (!e.isUnmodifiableByOtherCode()) {
+                            // update local variables
+                            localVariableValues.keySet().remove(e);
+                            // update field values
+                            localVariableValues.keySet().remove(e);
```

Comment:
This line is identical to the one above.  Should it be `fieldValues`?  First, add a test case that fails with the current code, then fix the code.